lib/wallet/account/hot-wallet-private-key.ts

```diff
@@ -116,29 +120,50 @@ export class HotWalletPrivateKey implements IWallet {
       throw new Error("No transaction to sign");
     }
 
-    // Replace the input script with the provided script
-    if (options?.scriptHex) {
-      const replaceIndex = pending.transaction.inputs.findIndex(
-        (input) => input.signatureScript === "",
-      );
-      if (replaceIndex === -1) {
-        throw new Error("No input to replace");
+    const signed = await this.signTx(pending.transaction, options?.scripts);
+    return (await this.rpcClient.submitTransaction({ transaction: signed }))
+      .transactionId;
+  }
+
+  // NOTE: This method does not support signing with multiple keys
+  async signTx(tx: Transaction, scripts?: ScriptOption[]) {
+    if (scripts) {
+      for (const script of scripts) {
+        await this.signTxInputWithScript(tx, script);
       }
+      return await signTransaction(tx, [this.getPrivateKey()], false);
+    }
+    return await signTransaction(tx, [this.getPrivateKey()], false);
+  }
 
-      const scriptBuilder = ScriptBuilder.fromScript(options.scriptHex);
-      const signature = await pending.createInputSignature(
-        replaceIndex,
-        new PrivateKey(this.getPrivateKey()),
-      );
+  async signTxWithScripts(tx: Transaction, scripts: ScriptOption[]) {
+    for (const script of scripts) {
+      await this.signTxInputWithScript(tx, script);
+    }
+  }
+
+  async signTxInputWithScript(tx: Transaction, script: ScriptOption) {
+    // check if the input does exist
+    if (tx.inputs.length <= script.inputIndex) {
+      throw new Error("Input index out of range");
+    }
 
-      pending.fillInput(
-        replaceIndex,
-        scriptBuilder.encodePayToScriptHashSignatureScript(signature),
-      );
+    // check if the input is not already signed
+    if (tx.inputs[script.inputIndex].signatureScript !== "") {
+      throw new Error("Input already signed");
     }
 
-    await pending.sign([this.getPrivateKey()], false);
-    return await pending.submit(this.rpcClient);
+    const signature = await createInputSignature(
```

Comment:
No need to await function does not return a promise.

Reply:
https://github.com/forbole/kastle/pull/22/commits/a6e327cb9da81bd11877b8d2167946adf9741c98